--- a/drivers/net/ethernet/Kconfig	2019-01-17 20:50:59.693143039 +0800
+++ b/drivers/net/ethernet/Kconfig	2019-01-17 20:53:12.669143039 +0800
@@ -14,6 +14,7 @@ if ETHERNET
 
 config MDIO
 	tristate
+	default y
 
 config SUNGEM_PHY
 	tristate
--- a/drivers/net/phy/mdio-bitbang.c	2019-01-18 21:22:53.018001000 +0800
+++ b/drivers/net/phy/mdio-bitbang.c	2019-01-18 21:24:51.198001000 +0800
@@ -14,6 +14,7 @@
  * Vitaly Bordug <vbordug@ru.mvista.com>
  */
 
+#include <linux/irqflags.h>
 #include <linux/module.h>
 #include <linux/mdio-bitbang.h>
 #include <linux/types.h>
@@ -153,7 +154,9 @@ static int mdiobb_read(struct mii_bus *b
 {
 	struct mdiobb_ctrl *ctrl = bus->priv;
 	int ret, i;
+	long flags;
 
+	local_irq_save(flags);
 	if (reg & MII_ADDR_C45) {
 		reg = mdiobb_cmd_addr(ctrl, phy, reg);
 		mdiobb_cmd(ctrl, MDIO_C45_READ, phy, reg);
@@ -178,13 +181,17 @@ static int mdiobb_read(struct mii_bus *b
 
 	ret = mdiobb_get_num(ctrl, 16);
 	mdiobb_get_bit(ctrl);
+	local_irq_restore(flags);
+
 	return ret;
 }
 
 static int mdiobb_write(struct mii_bus *bus, int phy, int reg, u16 val)
 {
 	struct mdiobb_ctrl *ctrl = bus->priv;
-
+	long flags;
+ 
+	local_irq_save(flags);
 	if (reg & MII_ADDR_C45) {
 		reg = mdiobb_cmd_addr(ctrl, phy, reg);
 		mdiobb_cmd(ctrl, MDIO_C45_WRITE, phy, reg);
@@ -199,6 +206,8 @@ static int mdiobb_write(struct mii_bus *
 
 	ctrl->ops->set_mdio_dir(ctrl, 0);
 	mdiobb_get_bit(ctrl);
+	local_irq_restore(flags);
+
 	return 0;
 }
 
